### PR TITLE
chore(merge): resolve merge conflicts from main

### PR DESCRIPTION
**Pull request by annawen1**
### Related Ticket(s)

[[Project]: merge main into feat/cwc-v2#10673](https://github.com/carbon-design-system/carbon-for-ibm-dotcom/issues/10673)

### Description

Resolve merge conflicts from `main` to `feat/cwc-v2` branch

### CODE REVIEW

File: packages/web-components/src/components/masthead/__stories__/masthead.stories.ts [hunk 1]
```diff
@@ -115,46 +92,30 @@ export const Default = (args) => {
     ${useMock
       ? html`
           <dds-masthead-composite
-            platform="${ifDefined(platform)}"
-            .platformUrl="${ifDefined(platformData.url)}"
-            selected-menu-item="${ifDefined(selectedMenuItem)}"
-            user-status="${ifDefined(userStatus)}"
-            searchPlaceholder="${ifDefined(searchPlaceholder)}"
-            .authenticatedProfileItems="${ifDefined(authenticatedProfileItems)}"
+            selected-menu-item="${ifNonNull(selectedMenuItem)}"
```
Comment (ariellalgilmore) [inline packages/web-components/src/components/masthead/__stories__/masthead.stories.ts:95]: mmissing `ifNonNull` import: https://github.com/carbon-design-system/carbon-for-ibm-dotcom/actions/runs/5556085716/jobs/10148137298?pr=10679

Comment (annawen1) [reply packages/web-components/src/components/masthead/__stories__/masthead.stories.ts]: good catch!

File: packages/web-components/src/components/masthead/top-nav-menu.ts [hunk 2]
```diff
@@ -36,7 +31,7 @@ class DDSTopNavMenu extends BXHeaderMenu {
   updated(changedProperties) {
     super.updated(changedProperties);
     if (changedProperties.has('active')) {
-      const { active, _triggerNode: triggerNode } = this;
+      const { active, _topMenuItem: triggerNode } = this;
```
Comment (ariellalgilmore) [inline packages/web-components/src/components/masthead/top-nav-menu.ts:34]: should this be `_triggerNode` ?

File: packages/web-components/src/components/masthead/megamenu-top-nav-menu.ts [hunk 3]
```diff
@@ -26,16 +29,16 @@ const { stablePrefix: ddsPrefix } = ddsSettings;
...
+  megaMenu!: DDSMegaMenu;
 
   /**
-   * The trigger button.
+   * The menu ul node.
    */
-  @query('[part="trigger"]')
-  private _topMenuItem!: HTMLAnchorElement;
+  @query(`.${prefix}--header__menu`)
+  private _menuNode!: HTMLElement;
```
Comment (ariellalgilmore) [inline packages/web-components/src/components/masthead/megamenu-top-nav-menu.ts:42]:
```suggestion
  /**
   * The trigger button.
   */
  @query('[part="trigger"]')
  private _topMenuItem!: HTMLAnchorElement;
```

Comment (annawen1) [reply packages/web-components/src/components/masthead/megamenu-top-nav-menu.ts:42]:
```
/**
   * The menu ul node.
   */
  @query(`.${prefix}--header__menu`)
  private _menuNode!: HTMLElement;
```

is what I see currently from the `main` branch, was there a change in the `feat/cwc-v2` branch for megamenu?

File: packages/web-components/src/components/masthead/top-nav-menu.ts (hunk 2 above)
Comment (annawen1) [reply packages/web-components/src/components/masthead/top-nav-menu.ts:34]: yeah it's coming from some changes that the AEM team made 4 weeks ago, so we should keep `_topMenuItem`

**Review by ariellalgilmore — APPROVED** [review]
Noicceee!

**Pull request merged**